e2e/playwright/list-views/src/tests/file-libraries.spec.ts

```diff
@@ -0,0 +1,168 @@
+      await siteActionsAdmin.addSiteMember(adminSite1, username, Site.RoleEnum.SiteConsumer);
+      await siteActionsAdmin.addSiteMember(adminSite2, username, Site.RoleEnum.SiteContributor);
+      await siteActionsAdmin.addSiteMember(adminSite3, username, Site.RoleEnum.SiteCollaborator);
+      await siteActionsAdmin.addSiteMember(adminSite4, username, Site.RoleEnum.SiteManager);
+      await siteActionsAdmin.addSiteMember(adminSite5, username, Site.RoleEnum.SiteConsumer);
+
+      await favoritesActions.addFavoriteById('site', adminSite1);
+      await favoritesActions.addFavoriteById('site', adminSite2);
+      await favoritesActions.addFavoriteById('site', adminSite3);
+      await favoritesActions.addFavoriteById('site', adminSite4);
+
+      await siteActionsUser.createSite(siteName, Site.VisibilityEnum.PUBLIC, null, siteId1);
+      await siteActionsUser.createSite(siteName, Site.VisibilityEnum.PUBLIC, null, siteId2);
+    } catch (error) {
+      console.error(`----- beforeAll failed : ${error}`);
+    }
+  });
+
+  test.afterAll(async () => {
+    await siteActionsUser.deleteSites([userSitePublic, userSiteModerated, userSitePrivate, siteId1, siteId2]);
+    await siteActionsAdmin.deleteSites([adminSite1, adminSite2, adminSite3, adminSite4, adminSite5]);
+  });
+
+  test.describe('My Libraries', () => {
+    test.beforeEach(async ({ page, myLibrariesPage }) => {
+      const loginPage = new LoginPage(page);
+      await loginPage.loginUser(
+        { username, password: username },
+        {
+          withNavigation: true,
+          waitForLoading: true
+        }
+      );
+      await myLibrariesPage.navigate();
+    });
+
+    test('[C217095] has the correct columns', async ({ myLibrariesPage }) => {
+      const expectedColumns = ['Name', 'Description', 'My Role', 'Visibility'];
+      const actualColumns = await myLibrariesPage.dataTable.getColumnHeaders();
+      expect(actualColumns).toEqual(expectedColumns);
+    });
+
+    test('[C289905] Library visibility is correctly displayed', async ({ myLibrariesPage }) => {
+      const expectedSitesVisibility = {
+        [userSitePrivate]: Site.VisibilityEnum.PRIVATE,
+        [userSiteModerated]: Site.VisibilityEnum.MODERATED,
+        [userSitePublic]: Site.VisibilityEnum.PUBLIC
+      };
+      for (const [site, visibility] of Object.keys(expectedSitesVisibility)) {
+        const sitesVisibility = await myLibrariesPage.dataTable.getRowAllInnerTexts(site);
+        expect(sitesVisibility.toLowerCase()).toContain(visibility.toLowerCase());
+      }
+    });
+
+    test('[C289903] User role is correctly displayed', async ({ myLibrariesPage }) => {
+      const expectedSitesRoles = {
+        [adminSite1]: Site.RoleEnum.SiteConsumer,
+        [adminSite2]: Site.RoleEnum.SiteContributor,
+        [adminSite3]: Site.RoleEnum.SiteCollaborator,
+        [adminSite4]: Site.RoleEnum.SiteManager
+      };
+
+      for (const [site, role] of Object.keys(expectedSitesRoles)) {
```

Comment:
@akashrathod28 you put const [site, role] as I suggested but you forgot to use Object.entries instead of Object.keys. To make it working you need to use Object.entries